### PR TITLE
Fix natp implementation. Add pairs.

### LINKED ISSUE #1: No luck with appendix code - bug with J-Bob-step?
Hey @holguinj,

I'm working through the appendix in the Little Prover, and the first few examples aren't simplifying for me. It looks like some issue with the paths code. I ran the following snippets in `clj-bob.repl`.

This works great:

``` scheme
(J-Bob-step (prelude)
            '(car (cons 'ham '(cheese)))
            '((() (car-cons 'ham '(cheese)))))
```

The code returns `'ham`, as expected. Once I add a path in to a new focus, however, the simplification rules stop working:

``` scheme
(J-Bob-step (prelude)
            '(equal 'flapjack (atom (cons a b)))
            '(((2) (atom-cons a b))
              (() (equal 'flapjack 'nil))))
```

This code just returns the original expression, `'(equal 'flapjack (atom (cons a b)))`. If I replace the `(2)` with an empty list and un-nest the `(atom ,,,)` snippet, that section gets simplified to `nil`.

Same problem with the second example:

``` scheme
(J-Bob-step (prelude)
            '(atom (cdr (cons (car (cons p q)) '())))
            '(((1 1 1) (car-cons p q))
              ((1) (cdr-cons p '()))
              (() (atom '()))))
```

Even a single step with a path specified seems to fail. Am I missing something obvious here?

Thanks so much for the awesome port!

### PR DESCRIPTION
**Pull request by sritchie**
This fixes #1, and allows you to try out the examples in chapters 9 and 10.

### CODE REVIEW

File: src/clj_bob/lang.clj [hunk 1]
```diff
@@ -67,7 +82,7 @@
 
 (defn nat? [x]
   (if (and (integer? x)
-           (< x 0))
+           (< 0 x))
```
Comment (sritchie) [inline src/clj_bob/lang.clj:85]: here it was!

Comment (juxtin) [inline src/clj_bob/lang.clj:85]: d'oh! nice catch.

**Comment by sritchie** [comment]
@holguinj any chance you want to merge this into master?

**Comment by juxtin** [comment]
Sorry about the delay! Thanks a lot for fixing this, @sritchie.

**Pull request merged**